x-pack/plugins/security/server/config.ts

```diff
@@ -314,13 +314,40 @@ export const ConfigSchema = schema.object({
       roleMappingManagementEnabled: schema.boolean({ defaultValue: true }),
     }),
   }),
+  experimental: schema.object({
+    fipsMode: schema.object({
+      enabled: schema.boolean({ defaultValue: false }),
+    }),
+  }),
 });
 
+function checkFipsConfig(config: RawConfigType, logger: Logger) {
+  const isFipsEnabled = config.experimental.fipsMode.enabled;
+  const isNodeRunningWithFipsEnabled = getFips() === 1;
+
+  // Check if FIPS is enabled in either setting
+  if (isFipsEnabled || isNodeRunningWithFipsEnabled) {
+    // FIPS must be enabled on both or log and error an exit Kibana
+    if (isFipsEnabled !== isNodeRunningWithFipsEnabled) {
+      logger.error(
+        `Configuration mismatch error. xpack.security.experimental.fipsMode.enabled is set to ${isFipsEnabled} and the configured Node.js environment has FIPS ${
+          isNodeRunningWithFipsEnabled ? 'enabled' : 'disabled'
+        }`
+      );
+      process.exit(78);
```

Comment:
🙈  I gonna do like I did not saw a plugin's configuration validation call `process.exit` with a return code not used anywhere else in of our software.

More seriously though, can't this be done at Core's level? I would have way less concerns and issues with Core performing this check (or at least this non-standard process termination). This is done before the license is added to the equation, right? So from my understanding nothing blocks it from being done in Core?

(but if we can't I'll just 🙈)

Reply:
😅 fair points, I'll take a look to see if there might be a better place for this check in core.

Did you have any place in mind that might be best?

Reply:
Could this possibly be done in the `setup` phase of Core's `security` service (where you added the logic to return the `isFIPS` API from the contract)?

Reply:
Done in https://github.com/elastic/kibana/pull/181187/commits/2c116105b50bce6079b893ed0029b8808607d534